### PR TITLE
Filter invalid XML response characters before parsing, and hard-fail parse errors.

### PR DESCRIPTION
**Pull request by amcnamara**
Fixes https://github.com/getoutreach/outreach-issues/issues/1464

### CODE REVIEW

File: spec/soap_data/find_folder_malformed_response.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+                        xmlns:t="http://schemas.microsoft.com/exchange/services/2006/types" 
+                        xmlns="http://schemas.microsoft.com/exchange/services/2006/messages">
+      <m:ResponseMessages>
+        <m:FindFolderResponseMessage ResponseClass="Success">
+          <m:ResponseCode>NoError</m:ResponseCode>
+          <m:RootFolder TotalItemsInView="1" IncludesLastItemInRange="true">
+            <t:Folders>
+              <t:Folder>
+                <t:FolderId Id="AQAnAH" ChangeKey="AQAAABY" />
+                <t:DisplayName>DING!</t:DisplayName>
```
Comment (amcnamara) [inline spec/soap_data/find_folder_malformed_response.xml:20]: There is a non-printing `bell` character in here.

File: lib/ews/soap/parsers/ews_parser.rb [hunk 2]
```diff
@@ -28,22 +28,24 @@ def initialize(soap_resp)
...
-      sax_parser.parse(sanitize_response(@soap_resp))
+      sax_parser.parse(clean_xml(@soap_resp))
       opts[:response_class].new @sax_doc.struct
     end
 
-    #
-    # We empirically encountered a &#x10; in the wild which is an invalid XML character.
-    # In order to parse correctly we need to remove
-    #
-    def sanitize_response(resp)
```
Comment (amcnamara) [inline lib/ews/soap/parsers/ews_parser.rb:39]: Didn't realize this was in here until I rebased against our branch, it's a serious bummer that this project isn't actively maintained.

**Comment by ghempton** [comment]
LGTM, forget we even had that sanitization code. Feel free to merge.

**Comment by ghempton** [comment]
Is there an accompanying PR to update our main repo's gemfile?

**Comment by amcnamara** [comment]
@ghempton I needed to wait until we merged to figure out the revision ID to peg the app repo to; I'll do that now.

**Pull request merged**